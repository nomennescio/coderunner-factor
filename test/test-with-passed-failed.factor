! Copyright 2022 nomennescio

USING: accessors debugger io kernel namespaces sequences tools.testest ;
IN: tests

: pass1 ( -- ) <{ "Hello World!" -> "Hello World!" }> ;
: pass2 ( -- ) <{ "Hello Worlds!" -> "Hello Worlds!" }> ;
: fail1 ( -- ) <{ "Hello World!" -> "Hello Worlds!" }> ;
: fail2 ( -- ) <{ "Hello Worlds!" -> "Hello Worlds! " }> ;

: run-test ( -- )
  "short strings" it#{ pass1 pass2 }#
  "failing compares" it#{ fail1 fail2 }#
;

: custom1 ( -- ) [ "Just passed" write ] [ "Just failed" write drop ] [ run-test ] with-passed-failed ;
: custom2 ( -- ) [ "Test passed" write ] [ error. ] [ run-test ] with-passed-failed ;
: custom3 ( -- ) run-test ;

: run-tests ( -- )

  "custom message" describe#{
    custom1
    custom2
    custom3
  }#
  
  "nested custom messages" describe#{

    [ "Test passed A" write ] [ "Test failed A " write error. ] [
      "short strings" it#{
        [ "Passed 1" write ] [ pass1 ] with-passed
        [ "Passed 2" write ] [ pass2 ] with-passed
      }#
      
      "failing compares" it#{
        [ "Passed 3" write ] [ fail1 ] with-passed
        [ "Passed 4" write ] [ fail2 ] with-passed
      }#
    ] with-passed-failed
    
    [ "Test passed B" write ] [ "Test failed B " write error. ] [
      "short strings" it#{
        [ "Failed 1" write drop ] [ pass1 ] with-failed
        [ "Failed 2" write drop ] [ pass2 ] with-failed
      }#
      
      "failing compares" it#{
        [ "Failed 3" write drop ] [ fail1 ] with-failed
        [ "Failed 4" write drop ] [ fail2 ] with-failed
      }#
    ] with-passed-failed

  }#
;

MAIN: run-tests