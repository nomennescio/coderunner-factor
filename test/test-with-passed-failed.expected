
<DESCRIBE::>custom message

<IT::>short strings

<PASSED::>Just passed

<PASSED::>Just passed

<COMPLETEDIN::>0.066171 ms

<IT::>failing compares

<FAILED::>Just failed

<FAILED::>Just failed

<COMPLETEDIN::>0.057857 ms

<IT::>short strings

<PASSED::>Test passed

<PASSED::>Test passed

<COMPLETEDIN::>0.018855 ms

<IT::>failing compares

<FAILED::>Expected :<:LF:>"Hello Worlds!"<:LF:>but got :<:LF:>"Hello World!"

<FAILED::>Expected :<:LF:>"Hello Worlds! "<:LF:>but got :<:LF:>"Hello Worlds!"

<COMPLETEDIN::>0.131308 ms

<IT::>short strings

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.019944 ms

<IT::>failing compares

<FAILED::>Test Failed : Expected :<:LF:>"Hello Worlds!"<:LF:>but got :<:LF:>"Hello World!"

<FAILED::>Test Failed : Expected :<:LF:>"Hello Worlds! "<:LF:>but got :<:LF:>"Hello Worlds!"

<COMPLETEDIN::>0.060291 ms

<COMPLETEDIN::>0.540606 ms

<DESCRIBE::>nested custom messages

<IT::>short strings

<PASSED::>Passed 1

<PASSED::>Passed 2

<COMPLETEDIN::>0.031492 ms

<IT::>failing compares

<FAILED::>Test failed A Expected :<:LF:>"Hello Worlds!"<:LF:>but got :<:LF:>"Hello World!"

<FAILED::>Test failed A Expected :<:LF:>"Hello Worlds! "<:LF:>but got :<:LF:>"Hello Worlds!"

<COMPLETEDIN::>0.044357 ms

<IT::>short strings

<PASSED::>Test passed B

<PASSED::>Test passed B

<COMPLETEDIN::>0.018118 ms

<IT::>failing compares

<FAILED::>Failed 3

<FAILED::>Failed 4

<COMPLETEDIN::>0.035839 ms

<COMPLETEDIN::>0.151897 ms
